server/events/checks_enabled_approve_policies_command_runner.go

```diff
@@ -0,0 +1,103 @@
+package events
+
+import (
+	"fmt"
+
+	"github.com/runatlantis/atlantis/server/events/command"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/lyft/feature"
+)
+
+type ChecksEnabledApprovePoliciesCommandRunner struct {
+	command.Runner
+
+	PrjCommandRunner  ProjectPolicyCheckCommandRunner
+	PrjCommandBuilder ProjectPlanCommandBuilder
+	FeatureAllocator  feature.Allocator
+}
+
+func NewChecksEnabledApprovePoliciesCommandRunner(runnner command.Runner, featureAllocator feature.Allocator, prjCommandRunner ProjectPolicyCheckCommandRunner, prjCommandBuilder ProjectPlanCommandBuilder) command.Runner {
+	return &ChecksEnabledApprovePoliciesCommandRunner{
+		Runner:            runnner,
+		FeatureAllocator:  featureAllocator,
+		PrjCommandRunner:  prjCommandRunner,
+		PrjCommandBuilder: prjCommandBuilder,
+	}
+}
+
+func (c ChecksEnabledApprovePoliciesCommandRunner) Run(ctx *command.Context, cmd *command.Comment) {
+	if !c.isChecksEnabled(ctx, ctx.HeadRepo, ctx.Pull) {
+		c.Runner.Run(ctx, cmd)
+		return
+	}
+
+	prjCmds, err := c.PrjCommandBuilder.BuildPlanCommands(ctx, &command.Comment{
+		RepoRelDir:  cmd.RepoRelDir,
+		Name:        command.Plan,
+		Workspace:   cmd.Workspace,
+		ProjectName: cmd.ProjectName,
+		LogLevel:    cmd.LogLevel,
+	})
+	if err != nil {
+		ctx.Log.WarnContext(ctx.RequestCtx, fmt.Sprintf("unable to build plan command: %s", err))
+
+		// Instead of hard failing, we log and forward the req and let the next runner fail the operation
+		c.Runner.Run(ctx, cmd)
+		return
+	}
+
+	// ApprovePoliciesCommandRunner uses this output for updating the github status check output
+	cmd.PolicyCheckOutput = c.getOutputForFailingPolicies(ctx, prjCmds)
```

Comment:
Why does this have to be here instead of directly in ApprovePolicyCommandRunner? or better yet, why not just expose a method from there that you can call here with the output, instead of confusingly wrapping things and mutating data models

Reply:
You mean have something along the lines of 
```
result := PolicyCheckOutputGenerator.GenerateOuptut(...., result) 
```
in approvePoliciesCommandRunner and if checks is enabled, we populate the policy check output for failing policies?

Reply:
Or, one other idea I have is wrapping 
```
type ProjectApprovePoliciesCommandRunner interface {
	// Approves any failing OPA policies.
	ApprovePolicies(ctx command.ProjectContext) command.ProjectResult
}
```

and basically running policy check and populating the output. This was the command runner won't have to be aware if checks is enabled or not.

Reply:
the first idea is fine, no need to over abstract.